api/python/tests/integration/test_packages.py

```diff
@@ -666,6 +666,61 @@ def test_s3_set_dir(self):
 
             list_object_versions_mock.assert_called_with('bucket', 'foo/')
 
+    def test_set_dir_update_policy(self):
+        """Verify building a package with update policy. """
+
+        # Test for local src dir
+        foodir = pathlib.Path("foo_dir")
+        bazdir = pathlib.Path("baz_dir")
+        foodir.mkdir()
+        bazdir.mkdir()
+        with open(foodir / 'bar', 'w') as fd:
+            fd.write(fd.name)
+        with open(bazdir / 'bar', 'w') as fd:
+            fd.write(''.join([str(i) for i in range(100)]))
+
+        pkg = Package()
+        pkg = pkg.set_dir("/", ".", meta={'name': 'test_meta'})
+        assert 'foo_dir' in pkg.keys()
+        existing_bar_size = pkg['foo_dir/bar'].size
+
+        # existing update policy
+        pkg = pkg.set_dir("foo_dir", bazdir, update_policy='existing')
+        assert 'foo_dir' in pkg.keys()
+        assert existing_bar_size == pkg['foo_dir/bar'].size
```

Comment:
We should really check metadata and hashes to be certain things haven't changed at all. In fact, copy the PackageEntry and check for equality?

Reply:
The test should check the entire state of the in-memory package is my main point; so whatever it takes to get there.